### PR TITLE
Release/1.2

### PR DESCRIPTION
**Pull request by QuintinWillison**
Make 1.2 the current version, ensuring that 1.1 versions are accessible (where applicable).

### CODE REVIEW

**Comment by QuintinWillison** [comment]
Thanks @SimonWoolf - I guess this ties in with #845 which I think I'll get merged first anyway, given it amends docs repository state for 1.1.

What goes first at this stage in terms of getting those realtime changes made, if in light of @mattheworiordan's #845 PR they still need to be made? Can we get this release merged to [bleeding edge docs](https://docs.ably.io/) via `master` branch or should the realtime changes be made first, if they are to be made at all? And if they are not going to be made then presumably I need to revert some of the changes I made in this PR.

**Comment by tcard** [comment]
@QuintinWillison Shouldn't this be deployed to docs.ably.io?

**Comment by QuintinWillison** [comment]
Hi @tcard. Yes, it's live. Happens automatically once it lands on `master` branch.

**Comment by tcard** [comment]
I commented because I was seeing 1.1 on docs.ably.io, but refreshing fixed that, so I guess it was cached or something. Sorry for the noise.

**Comment by SimonWoolf** [comment]
As discussed [on slack](https://ably-real-time.slack.com/archives/C030C5YLY/p1585732884044400?thread_ts=1585608426.028800&cid=C030C5YLY) the diff is wrong, looks like it's from `1.1-as-amended` to `1.2` when it should be from `1.1` to `1.2` - ie should include all the changes that have been added to the 1.1 spec since its initial release, which are required for 1.2 compliance per https://github.com/ably/docs/#branch-and-tag-scheme-for-features-spec.

Apologies for not picking up on this during my PR review on monday

File: content/general/events/ifttt.textile [hunk 1]
```diff
@@ -65,7 +65,7 @@ x-ably-envelope-source: channel.message
 x-ably-message-encoding: json
 x-ably-message-id: {UNIQUE_ABLY_MESSAGE_ID}
 x-ably-message-timestamp: {TIMESTAMP_ORIGINAL_MESSAGE_WAS_SENT}
-x-ably-version: 1.0
+x-ably-version: 1.2
```
Comment (mattheworiordan) [inline content/general/events/ifttt.textile:68]: @SimonWoolf is this correct?  Will the Reactor rules use a 1.2 now?

File: content/partials/general/events/_batched_event_headers.textile [hunk 2]
```diff
@@ -3,4 +3,4 @@ Batched events will have the following headers:
 - content-type := the type of the payload. This can be either @application/json@, @text/plain@, or @application/octet-stream@, depending on if it's @JSON@, @text@, or @binary@ respectively
 - x-ably-envelope-appid := the "app ID":https://support.ably.io/support/solutions/articles/3000063083 which the message came from
 - content-type := the type of the payload. This will be @application/json@ or @application/x-msgpack@
-- x-ably-version := the version of Reactor Event. At present this should be @1.0@, though older Events will be @0.8@
+- x-ably-version := the version of Reactor Event. At present this should be @1.2@
```
Comment (mattheworiordan) [inline content/partials/general/events/_batched_event_headers.textile:6]: As above...

File: content/realtime/channel-params.textile [hunk 3]
```diff
@@ -57,7 +57,7 @@ h3. SSE example
 For example, to specify the @rewind@ channel param with the value @"1"@ using a querystring parameter, where it will apply to all channels:
 
 ```[javascript]
-var querystring = 'v=1.1&channels={{RANDOM_CHANNEL_NAME}}&rewind=1&key={{API_KEY}}';
+var querystring = 'v=1.2&channels={{RANDOM_CHANNEL_NAME}}&rewind=1&key={{API_KEY}}';
```
Comment (mattheworiordan) [inline content/realtime/channel-params.textile:60]: Does SSE suport 1.2?

**Comment by mattheworiordan** [comment]
I am afraid we have an issue here with this PR in that we've effectively skipped a version now in the docs:

screenshot_2020-04-01_11-28-54_pm

This is important because if someone is now using 1.1, what docs do they refer to? The process we followed before was to simply copy the files into a 1.1 versions folder, and voila, you magically have 1.1.

We need to do this I am afraid.

Also, this has been merged in `master`, which means if anyone now merges other changes in the docs repo (@fliptopbox is doing work on tutorials), we will now be showing 1.2 on the website.  Did we intend to do that before we actually release 1.2 libraries?

What's the plan for release of all 1.2 libraries too i.e. we should bump all libraries I assume to 1.2 even if they don't necessarily change so that we have a set of 1.2 libraries, and the features doc is updated to confirm what features are missing from which libs for 1.2 specific things.

File: content/general/events/ifttt.textile (hunk 1 above)
Comment (SimonWoolf) [reply content/general/events/ifttt.textile:68]: Not yet, as I mentioned in [my review of this pr](https://github.com/ably/docs/pull/843#pullrequestreview-384134783). I'm planning to change the default for new rules, per the new versioning policy you wrote. (I still think this change is unnecessary and that changing the version without there being any actual difference is confusing to customers, but I've already lost that argument 🙂)

File: content/realtime/channel-params.textile (hunk 3 above)
Comment (SimonWoolf) [reply content/realtime/channel-params.textile:60]: Not yet, as I mentioned in [my review of this pr](https://github.com/ably/docs/pull/843#pullrequestreview-384134783)

**Comment by QuintinWillison** [comment]
@mattheworiordan can you please be specific regarding your statement "we've effectively skipped a version now in the docs". For what I was able to work out from inspecting the existing repository, the commits going towards the previous 1.1 release by @tomczoink, the Ruby 'versions helper' code in this repository and from adhoc input I've had from @SimonWoolf my work to get us to a state of having docs ready for 1.2 was complete. So what did I miss?

Your example screenshot above is from a page that did not need to change between 1.0 and 1.2 docs - so why do assert it needs an entry in the drop down for 1.1?

Screenshot 2020-04-06 at 06 46 53

The only two pages that have changed are channel params:

Screenshot 2020-04-06 at 06 47 17

and the features spec:

Screenshot 2020-04-06 at 06 47 04

I don't know what more work needs doing on this repository so can you either be more specific about what work still remains or perhaps push some commits yourself, please?

FYI, @paddybyers

**Comment by mattheworiordan** [comment]
Moving this conversation out of here as there is an open issue https://github.com/ably/docs/issues/854

**Pull request merged**